2019-02-11  Adam Christopher Cole  <adamc@lnxsrv07.seas.ucla.edu>

        * lib/timespec.h (timespec_cmp): Change timespec_cmp to use direct comparison
	instead of Subtraction to avoid integer overflow cases with files created far apart
	
--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2019-02-11 01:16:12.804503001 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec == b.tv_sec)
+  {  return 0;  }
+  else if (a.tv_sec > b.tv_sec)
+  {  return 1;  }
+  else
+  {  return -1;  }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
